### ISSUE TITLE
Improve handling of git packages

### ISSUE THREAD

**Issue by fosskers** [fosskers/aura#812]
### Discussed in https://github.com/fosskers/aura/discussions/811

<div type='discussions-op-text'>

<sup>Originally posted by **sonicbhoc** August 15, 2023</sup>
So one thing that I've noticed is that I have to constantly tell Aura for each package that I don't want to view diffs for the build files. On top of this, there doesn't seem to be a way to automatically skip packages that haven't changed their build files. I'd love to see options to configure these (and if they are there, I don't know how I missed them).

Finally, I noticed that some packages fail when copying files or when the build uses cmake due to path issues. I'm not sure exactly what causes these failures.

The packages that failed include:

* ofono-git
* uavs3d-git
* bluez-tools-git</div>

**Comment by fosskers** [comment]
@sonicbhoc Thank you for reporting! Are you using Aura 3 or Aura 4 when trying to build these?

**Comment by sonicbhoc** [comment]
Aura 4.

Also, I was going to mention that the incessant nagging for viewing the diffs was indeed a configuration option I accidentally enabled in the config toml file. Oops.

That being said, the rest of them are real issues. I'd love to be able to tell aura to skip building git packages if the build files didn't actually change after a pull. makepkg does it this way iirc; I don't recall it rebuilding things if nothing changed. I imagine such a feature could be added as something in the toml config file that can also be overridden with a command-line switch if necessary.

Lastly, the list of packages I encountered issues with (you can add bluez-git to the list, but bluez-alsa-git surprisingly works just fine) all seem to fail if the build has to do something with pathing or the local build directory somehow. Copying files and CMake list file failures seem to be common. When  I get home I can paste the output from Aura for you. Thanks!

**Comment by fosskers** [comment]
> makepkg does it this way iirc

Aura is calling `makepkg` internally, so this may be due to passing the `--force` flag that's making it do a build no matter what.

> When I get home I can paste the output from Aura for you. 

Yes this would be helpful, thanks! I will also try building on my own machine.

**Comment by fosskers** [comment]
In an attempt to build `ofono-git`:
```
10:41:10 [DEBUG] (1) aura::command::aur::build: Copying git://git.kernel.org/pub/scm/libs/ell/ell.git
```
Taking a look at it's PKGBUILD, we see:
```
source=("$pkgname::git+https://git.kernel.org/pub/scm/network/ofono/ofono.git"
  git://git.kernel.org/pub/scm/libs/ell/ell.git)
```
I believe the second entry this is improperly set. So this is an error by the package maintainer.

**Comment by fosskers** [comment]
Regarding `ofono-git`, I've [commented on its AUR page](https://aur.archlinux.org/packages/ofono-git#comment-979892) regarding the improperly set `source`. However, that package hasn't been updated for a few years, so you may wish to maintain a local copy.

**Comment by fosskers** [comment]
Regarding the others, I was able to build `uavs3d-git` and `bluez-tools-git` just now without issue. Can you confirm if the failures are still occurring for you?

**Comment by sonicbhoc** [comment]
> Regarding the others, I was able to build `uavs3d-git` and `bluez-tools-git` just now without issue. Can you confirm if the failures are still occurring for you?

I'm sorry, I wasn't getting notifications for this for some reason. I'll get back to you after I get done with my work day.

**Comment by fosskers** [comment]
By the way, it's now easier to install. https://aur.archlinux.org/packages/aura-git is updated to package Aura 4.

**Comment by sonicbhoc** [comment]
I haven't been using Arch as much recently, unfortunately. When I get my laptop running I'll definitely give it a go.

**Comment by fosskers** [comment]
This should already be what's happening, unless you force to rebuild with `--git`. What behaviour are you noticing at the moment?

**Comment by fosskers** [comment]
Right. Aura will automatically detect when a `-git` package on the AUR has actually updated its version, otherwise you need to pass `--git` to force a rebuild.

**Comment by kureta** [comment]
I did set `git = true` in the config file, thinking that it wouldn't check for `-git` package updates otherwise. Set it back to `false` now it seems to work normally, but I have to wait until a git package I use gets updated. 

The wording in the user guide "--git will consider all such packages for updates:" made me think that it will check all git packages for updates, as if normally it considers only some packages according to some internal logic or something. Wouldn't it be clearer to say "--git will **force** all such packages to be updated", or is it my comprehension skills? :)

**Comment by fosskers** [comment]
> --git will force all such packages to be updated

You're right, I'll alter the wording.

**Comment by fosskers** [comment]
> Would you consider adding such a mechanism?

You mean, auto-detection of whether the actual upstream package has new commits?
